scripts/create_project_table.py

```diff
@@ -0,0 +1,131 @@
+"""
+Process README files to extract metadata, generate a table of content, 
+and update a central README file with aggregated data.
+
+- Extract metadata from the top section of the README file for each project (e.g., between "---" delimiters).
+- Generate table rows summarizing the metadata
+- Insert the generated rows into the project table in the main README file
+
+Classes:
+- ReadmeMetadataProcessor: Extracts and processes metadata from individual README files.
+"""
+
+import os
+import yaml
+from pathlib import Path
+from typing import List
+
+ROOT_PATH = Path(__file__).parent.parent
+README_PATH = ROOT_PATH / "README.md"
+
+
+class ReadmeMetadataProcessor:
+    def __init__(self, readme_file: Path):
```

Comment:
Based on this `__init__` function, it looks like a class isn't really helping here a lot, there's no real state. You can just use module-scoped functions with parameters instead of class-scoped methods.

But feel free to wait a little with that, until we're done with the other comments.